src/main/java/de/hysky/skyblocker/skyblock/chocolatefactory/EggFinder.java

```diff
@@ -0,0 +1,146 @@
+package de.hysky.skyblocker.skyblock.chocolatefactory;
+
+import de.hysky.skyblocker.config.SkyblockerConfigManager;
+import de.hysky.skyblocker.utils.ColorUtils;
+import de.hysky.skyblocker.utils.Constants;
+import de.hysky.skyblocker.utils.waypoint.Waypoint;
+import it.unimi.dsi.fastutil.objects.ObjectImmutableList;
+import net.fabricmc.fabric.api.client.message.v1.ClientReceiveMessageEvents;
+import net.fabricmc.fabric.api.client.networking.v1.ClientPlayConnectionEvents;
+import net.fabricmc.fabric.api.client.rendering.v1.WorldRenderContext;
+import net.fabricmc.fabric.api.client.rendering.v1.WorldRenderEvents;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.component.DataComponentTypes;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.decoration.ArmorStandEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.text.Text;
+import org.apache.commons.lang3.mutable.MutableObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static de.hysky.skyblocker.skyblock.chocolatefactory.EggFinder.EggType.BREAKFAST;
+import static de.hysky.skyblocker.skyblock.chocolatefactory.EggFinder.EggType.DINNER;
+import static de.hysky.skyblocker.skyblock.chocolatefactory.EggFinder.EggType.LUNCH;
+
+public class EggFinder {
+	private static final Pattern eggFoundPattern = Pattern.compile("^(?:HOPPITY'S HUNT You found a Chocolate|You have already collected this Chocolate) (Breakfast|Lunch|Dinner)");
+	private static final Pattern newEggPattern = Pattern.compile("^HOPPITY'S HUNT A Chocolate (Breakfast|Lunch|Dinner) Egg has appeared!$");
+	private static final Logger logger = LoggerFactory.getLogger("Skyblocker Egg Finder");
+
+	private EggFinder() {
+	}
+
+	public static void init() {
+		ClientPlayConnectionEvents.JOIN.register((ignored, ignored2, ignored3) -> clearEggs());
+		ClientReceiveMessageEvents.GAME.register(EggFinder::onChatMessage);
+		WorldRenderEvents.AFTER_TRANSLUCENT.register(EggFinder::renderWaypoints);
+	}
+
+	public static void checkIfEgg(Entity entity) {
```

Comment:
I would limit this to only running if the player is on an island where eggs can spawn so that we aren't checking every entity, everywhere.

Maybe when the calendar data gets implemented in #691 we could also check for the event being active